game/hud/src/widgets/Social/components/OrdersGrid.tsx

```diff
@@ -0,0 +1,161 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ *
+ * @Author: Mehuge (mehuge@sorcerer.co.uk)
+ * @Date: 2017-03-08
+ */
+
+import * as React from 'react';
+import { graphql, InjectedGraphQLProps } from 'react-apollo';
+import { connect } from 'react-redux';
+import gql from 'graphql-tag';
+import { StyleSheet, css, StyleDeclaration } from 'aphrodite';
+import { Spinner, ColumnDefinition } from 'camelot-unchained';
+import GridViewPager from './GridViewPager';
+import GroupTitle from './GroupTitle';
+
+export interface OrdersGridStyle extends StyleDeclaration {
+  container : React.CSSProperties;
+}
+
+export const defaultOrdersGridStyle : OrdersGridStyle = {
+  container: {
+    flex: '1 1 auto',
+    display: 'flex',
+    flexDirection: 'column'
+  },
+};
+
+export interface OrdersGridQuery {
+  orders: {
+    totalCount: number;
+    data: {
+      id: string;
+      name: string;
+    }[]
+  }
+}
+
+export interface OrdersGridProps extends InjectedGraphQLProps<OrdersGridQuery> {
+  styles?: Partial<OrdersGridStyle>;
+  columnDefinitions?: ColumnDefinition[];
+  shard: number;
+  skip: number;
+  itemsPerPage: number;
+  filter: string;
+  gotoPage: (page: number) => void;
+  sort: string;
+  reverse: boolean;
+  orderBy: (by: string, asc: boolean) => void;
+};
+
+function stricmp(a: string, b: string) {
```

Comment:
we should just add these kind of things to utils in the cu lib so we don't have to re-write them in other places.